src/meta/Intersection.ts

```diff
@@ -0,0 +1,89 @@
+import global from '@dojo/shim/global';
+import WeakMap from '@dojo/shim/WeakMap';
+import Map from '@dojo/shim/Map';
+import { createHandle } from '@dojo/core/lang';
+import { Base } from './Base';
+
+import 'intersection-observer';
+
+interface ExtendedIntersectionObserverEntry extends IntersectionObserverEntry {
+	readonly isIntersecting: boolean;
+}
+
+interface IntersectionDetail extends IntersectionGetOptions {
+	entries: WeakMap<Element, IntersectionResult>;
+	observer: IntersectionObserver;
+}
+
+export interface IntersectionGetOptions {
+	root?: string;
+	rootMargin?: string;
+	threshold?: number[];
+}
+
+export interface IntersectionResult {
+	intersectionRatio: number;
+	isIntersecting: boolean;
+}
+
+const defaultIntersection: IntersectionResult = Object.freeze({
+	intersectionRatio: 0,
+	isIntersecting: false
+});
+
+export class Intersection extends Base {
+	private readonly _details: Map<string, IntersectionDetail> = new Map();
+
+	public get(key: string, options: IntersectionGetOptions = {}): IntersectionResult {
+		let rootNode: HTMLElement | undefined;
+		if (options.root) {
+			rootNode = this.getNode(options.root);
+			if (!rootNode) {
+				return defaultIntersection;
+			}
+		}
+		const node = this.getNode(key);
+		if (!node) {
+			return defaultIntersection;
+		}
+
+		let details = this._getDetails(options);
+		if (!details) {
+			details = this._createDetails(options, rootNode);
+			details.observer.observe(node);
+		}
+
+		return details.entries.get(node) || defaultIntersection;
+	}
+
+	public has(key: string | number, options?: IntersectionGetOptions): boolean {
+		const node = this.getNode(key);
+		const details = this._getDetails(options);
+		return Boolean(details && node && details.entries.has(node));
+	}
+
+	private _createDetails(options: IntersectionGetOptions, rootNode?: HTMLElement): IntersectionDetail {
+		const entries = new WeakMap<HTMLElement, ExtendedIntersectionObserverEntry>();
+		const observer = new global.IntersectionObserver(this._onIntersect(entries), options);
```

Comment:
The root string from the options object (instead of the root node) is passed when creating an IntersectionObserver. I have a test for this in:	 'intersections accept root'

Reply:
That makes a lot of sense, thanks (i'll add a test).

I actually tried the root node functionality yesterday but couldn't get it to work (just thought I had done it wrong.)